File: mpi/c/hello.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char *argv[])
{
    // Initialize the MPI library
    MPI_Init(&argc, &argv);

    // Determine how many processes will be run
    int world_size;
    MPI_Comm_size(MPI_COMM_WORLD, &world_size);

    int world_rank;
    MPI_Comm_rank(MPI_COMM_WORLD, &world_rank);

    char processor_name[256] = {0};
    int length = 256;
    MPI_Get_processor_name(processor_name, &length);

    printf("Hello world from rank %d out of %d processes, on node %s\n", world_rank, world_size, processor_name);

    // Shut down MPI
    MPI_Finalize();

    return 0;
}

File: mpi/python/hello.py
from mpi4py import MPI

world_size = MPI.COMM_WORLD.size
world_rank = MPI.COMM_WORLD.rank
name = MPI.Get_processor_name()

print(f"Hello world, from rank {world_rank} out of {world_size} processes (node name: {name})")
